tox.ini

```diff
@@ -92,7 +92,7 @@ commands =
 [testenv:integration]
 description = Run integration tests
 deps =
-    juju==2.9.44.0
+    juju==3.1.2.0
```

Comment:
We shouldn't be using 3.1.2, we should either use latest 2.9 or latest 3.1. The production instance is deployed on 2.9 so I suspect we should test against 2.9.45 (current latest 2.9).

Reply:
The lib does not have a later version of 3.1.x it seems, that's why I picked this one https://pastebin.canonical.com/p/j9tvYmZYRj/

Reply:
how about using `juju>=2,<3` so that we test on latest 2.9?

Reply:
I think it will be better to use juju 3, since the charm was recently created so we can stay more up to date with juju changes and there is no significant change to make test-wise.

Reply:
+1 for Yanks comment

If the idea is to change the juju version anyway, could the reason be added to the PR description then?

Reply:
The reason has been added to the PR description, thanks!